src/com/ConnectionInfoPublisher.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ConnectionInfoPublisher.hpp"
+#include <chrono>
+#include <thread>
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/string_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/filesystem.hpp>
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 int rank,
+                                                 std::string addressDirectory)
```

Comment:
Move-constructing strings is `noexcept`.

```suggestion
                                                 std::string addressDirectory) noexcept
```